### PR TITLE
Expose represent_mixins_as_columns as a public method

### PR DESCRIPTION
**Pull request by taldcroft**
From https://github.com/astropy/astropy/pull/7712#discussion_r209460176.

### CODE REVIEW

**Comment by taldcroft** [comment]
Now I'm wondering about going a step further and making this a Table method, with possible names like`flatten_mixins` or `encode_mixins` or <suggestions>??

It occurs to me that this would also be useful for:
- `to_pandas`
- `as_array` via an kwarg `flatten_mixins=True`.  This would be an API change, but currently for mixins it returns a column of object dtype with individual objects for each row.  Probably not that useful in most cases.  (Could also make the kwarg default to `False` for no API change).

File: astropy/table/serialize.py [hunk 1]
```diff
@@ -120,11 +120,59 @@ def _represent_mixin_as_column(col, name, new_cols, mixin_cols,
...
+    additional information needed to construct the original mixin columns from
+    the split columns.
+
+    This function is used by astropy I/O when writing tables to ECSV, FITS,
+    HDF5 formats.
+
+    Note that if the table does not include any mixin columns then the original
+    table is returned with no update to ``meta``.
+
+    Examples
```
Comment (mhvk) [inline astropy/table/serialize.py]: `Examples` should be the last section in a docstring...  http://docs.astropy.org/en/stable/development/docrules.html

File: astropy/table/serialize.py [hunk 2]
```diff
@@ -120,11 +120,59 @@ def _represent_mixin_as_column(col, name, new_cols, mixin_cols,
...
+    ----------
+    tbl : `~astropy.table.Table` or subclass
+        Table to represent mixins as Columns
+    exclude_classes : tuple of classes
+        Exclude any mixin columns which are subclasses of tuple elements
+
+    Returns
+    -------
+    tbl : `~astropy.table.Table`
+        Table with updated columns, or else the original input ``tbl``
```
Comment (mhvk) [inline astropy/table/serialize.py]: "updated columns" sounds like it is done in-place. Maybe explicitly "New table with ..."

File: astropy/table/serialize.py [hunk 3]
```diff
@@ -120,11 +120,59 @@ def _represent_mixin_as_column(col, name, new_cols, mixin_cols,
...
+    ------- ------- -------------- -------------- -------
+        1.0     3.0      2451180.0          -0.25   100.0
+        2.0     4.0      2451545.0            0.0   200.0
+
+    Parameters
+    ----------
+    tbl : `~astropy.table.Table` or subclass
+        Table to represent mixins as Columns
+    exclude_classes : tuple of classes
+        Exclude any mixin columns which are subclasses of tuple elements
```
Comment (mhvk) [inline astropy/table/serialize.py]: "...which are instances of any of the classes in the tuple"

**Comment by bsipocz** [comment]
This looks to be finished to me, but would need a rebase to make sure tests still pass. I'll do the rebase now, but plan to wait a few more hours for possible vetos before merging.

**Comment by taldcroft** [comment]
From #8284 and friends, it would be good to also make `Info._represent_as_dict()` and `Info._construct_from_dict` into public methods since they are now being used that way.

**Comment by bsipocz** [comment]
@mhvk - Could you please rereview? Ignore the bot, I'll fix the changelog before release.

**Comment by bsipocz** [comment]
Thank you @taldcroft and @mhvk!

**Comment by bsipocz** [comment]
Apparently rebasing is good practice to make sure stuff passed 4 month ago still passes. 

@mhvk - do you have time to address the failures we see on master? https://travis-ci.org/astropy/astropy/jobs/521494493#L2393

**Pull request merged**